services/uploads/src/clamAV.ts

```diff
@@ -0,0 +1,212 @@
+import path from 'path'
+import { spawnSync } from 'child_process'
+import { readdir } from 'fs/promises'
+
+import { S3UploadsClient } from './s3'
+
+type ClamAVScanResult = 'CLEAN' | 'INFECTED'
+
+interface ClamAV {
+    downloadAVDefinitions: () => Promise<undefined | Error>
+    uploadAVDefinitions: (workdir: string) => Promise<undefined | Error>
+    scanLocalFile: (path: string) => ClamAVScanResult | Error
+    fetchAVDefinitionsWithFreshclam: (workdir: string) => Promise<undefined | Error>
+}
+
+interface ClamAVConfig {
+    bucketName: string
+    definitionsPath: string
+    pathToClamav: string
+    pathToFreshclam: string
+    pathToConfig: string
+}
+
+function NewClamAV(config: Partial<ClamAVConfig>, s3Client: S3UploadsClient) {
+
+    if (!config.bucketName || !config.definitionsPath) {
+        throw new Error('BucketName and DefinitionsPath are required')
+    }
+
+    const fullConfig: ClamAVConfig = {
+        bucketName: config.bucketName,
+        definitionsPath: config.definitionsPath,
+
+        pathToClamav: config.pathToClamav || '/opt/bin/clamscan',
+        pathToFreshclam: config.pathToFreshclam || '/opt/bin/freshclam',
+        pathToConfig: config.pathToConfig || '/opt/bin/freshclam.conf'
+    }
+
+    return {
+        downloadAVDefinitions: () => downloadAVDefinitions(fullConfig, s3Client),
+        uploadAVDefinitions: (workdir: string) => uploadAVDefinitions(fullConfig, s3Client, workdir),
+        scanLocalFile: (path: string) => scanLocalFile(fullConfig, path),
+        fetchAVDefinitionsWithFreshclam: (workdir: string) => fetchAVDefinitionsWithFreshclam(fullConfig, workdir),
+    }
+}
+
+/**
+ * Uploads the AV definitions to the S3 bucket.
+ */
+async function uploadAVDefinitions(config: ClamAVConfig, s3Client: S3UploadsClient, workdir: string) {
+    // delete all the definitions currently in the bucket.
```

Comment:
I'd be interested for us to time this whole process of running `freshclam` to get definitions, clearing out the bucket, and then copying the new ones to the s3 bucket again. We could probably save some time by just calling the [s3 sync command](https://docs.aws.amazon.com/cli/latest/reference/s3/sync.html) with the CLI, which is sort of like `rsync` in that we'll only copy the diff. Not sure we'd want to re-implement that with the JS SDK though, probably just use the tool directly.

Maybe we add some timing logs in this PR, ship it, and then decide if it makes sense to do some follow on work.

Reply:
It’s only 3 or 4 files so I think it’s pretty quick. That could be sweet though, we’re syncing in both directions, uploading the new ones and downloading them for clamscan